webui/react/src/components/PaginatedNotesCard.tsx

```diff
@@ -0,0 +1,203 @@
+import { Button, Dropdown, Menu, Modal } from 'antd';
+import { SelectValue } from 'antd/lib/select';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import usePrevious from 'hooks/usePrevious';
+import Icon from 'shared/components/Icon/Icon';
+import { Note } from 'types';
+
+import NotesCard from './NotesCard';
+import css from './PaginatedNotesCard.module.scss';
+
+interface Props {
+  disabled?: boolean;
+  notes: Note[];
+  onDelete: (pageNumber: number) => void;
+  onNewPage: () => void;
+  onSave: (notes: Note[]) => void;
```

Comment:
**question:** below there are call patterns of `await onSave(...)`. Maybe this needs to be `(notes: Note[]) => Promise<void>`?